ngraph/test/util/unary_test.hpp

```diff
@@ -0,0 +1,293 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+#include "gtest/gtest.h"
+#include "ngraph/ngraph.hpp"
+#include "util/engine/test_engines.hpp"
+#include "util/test_case.hpp"
+#include "util/test_control.hpp"
+
+namespace ngraph
+{
+    namespace test
+    {
+        template <typename V>
+        struct ArgumentTolerance
+        {
+            ArgumentTolerance() = default;
+            ArgumentTolerance(V v)
+                : in_use{true}
+                , value{v}
+            {
+            }
+            bool in_use{false};
+            V value{};
+        };
+
+        using BitTolerance = ArgumentTolerance<int>;
+        using FloatTolerance = ArgumentTolerance<float>;
+
+        struct Tolerance
+        {
+            BitTolerance bit{};
+            FloatTolerance fp{};
+
+            Tolerance() = default;
+            Tolerance(const float tolerance)
+                : fp(tolerance)
+            {
+            }
+            Tolerance(const int tolerance)
+                : bit(tolerance)
+            {
+            }
+        };
+
+        template <typename TestEngine, ngraph::element::Type_t et>
+        class unary_test;
+
+        struct rand_normal_tag
+        {
+        };
+        struct rand_uniform_int_tag
+        {
+        };
+        struct rand_uniform_real_tag
+        {
+        };
+
+        template <ngraph::element::Type_t eleType>
+        class Data
+        {
+        public:
+            using T = ngraph::fundamental_type_for<eleType>;
+
+            // random (normal-distrubition) generate data
+            Data(rand_normal_tag, double mean, double stddev, const ngraph::Shape& s)
+                : value(ngraph::shape_size(s))
+                , shape(s)
+            {
+                std::mt19937 gen{0};
+                std::normal_distribution<> d{mean, stddev};
+                for (size_t i = 0; i < value.size(); ++i)
+                    value[i] = d(gen);
+            }
```

Comment:
I don't see reason to close generator inside constructors. If we need to extend this list in this approach we need to do allot of changes.

```suggestion
            template<typename ElementsGenerator>
            Data(ElementsGenerator g, const ngraph::Shape& s)
                : value(ngraph::shape_size(s))
                , shape(s)
            {
                for (size_t i = 0; i < value.size(); ++i)
                    value[i] = g();
            }
// or
            using ElementsGenerator = std::funtion<float()>;
            Data(ElementsGenerator g, const ngraph::Shape& s)
                : value(ngraph::shape_size(s))
                , shape(s)
            {
                for (size_t i = 0; i < value.size(); ++i)
                    value[i] = g();
            }
```

Reply:
Agree, I will change it using std::funtion<T()> as generator